### ISSUE TITLE
SiteMapUtilTest.testUpdateSiteMap failing in develop

### ISSUE THREAD

**Issue by donsizemore** [IQSS/dataverse#6854]
Appears unrelated to this morning's payara/openjdk funsies:

`[ERROR]   SiteMapUtilTest.testUpdateSiteMap:100 expected null, but was:<org.xml.sax.SAXParseException; schema_reference.4: Failed to read schema document 'https://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd', because 1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.>`

`[ERROR] Tests run: 1113, Failures: 1, Errors: 0, Skipped: 6`

/tmp/2013874958200619181/docroot/sitemap/sitemap.xml.staged for instance gets written, but then:

`SiteMapUtilTest.testUpdateSiteMap:96 expected null, but was:<java.io.FileNotFoundException: /tmp/2013874958200619181/docroot/sitemap/sitemap.xml (No such file or directory)>`

`[dls@payara5 dataverse]$ ls -l /tmp/2013874958200619181/docroot/sitemap/`

`-rw-rw-r-- 1 dls dls 412 Apr 23 09:00 sitemap.xml.staged`

**Comment by PaulBoon** [comment]
@donsizemore  I have the same problem, just tried to built our fork of 4.18.1
When browsing to https://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd
I get a NET::ERR_CERT_DATE_INVALID on the Chrome page, looks like the certificate has expired.

**Comment by pdurbin** [comment]
I just wanted to point out that validation also happens in the production code (SiteMapUtil) which has a couple implications:

- production installations will be affected (a workaround is to copy over the "staged" file which is generated before validation happens)
- if we refactor using the suggestion by @PaulBoon we should be sure to update not just the tests but the production code.

**Comment by donsizemore** [comment]
sitemaps.org updated their cert this morning, though the new cert expires soon. Leaving issue open in case we would like to include sitemap.xsd in the test resources folder.

**Comment by mdmADA** [comment]
Hi - is this issue 'fixed' now? I am trying to compile but getting this same error and I am not sure how to move past it.

Thanks!

**Comment by PaulBoon** [comment]
@mdmADA Its building on my PC now, but if if fails the test and you are sure you didn't change the code to make it fail you could always ignore it for the time being. Just put `@Ignore` above the line with `@Test`.

**Comment by PaulBoon** [comment]
@pdurbin I do remember that at some point we had a sitemap generation failure in production because the validation failed. That might have been the same thing if the certificate expired before. Also I do remember there was something with a staged file that needed to be removed manually after the failure. We might have to look into that.

**Comment by pdurbin** [comment]
Yes, assuming all is well in terms of certs and what not, the staged file needs to be removed after a failed run.

The idea with the staged file is that the production file stays in place until the staged file is properly created and validated.

I'm certainly happy to look at or talk about proposed changes to the code.

**Comment by mdmADA** [comment]
Thanks @pdurbin and @PaulBoon.

I deleted the staged file and it compiled. I had tried deleting the file previously and it hadn't compiled then but it does now.
